pvlib/test/test_pvgis.py

```diff
@@ -0,0 +1,182 @@
+    assert inputs_met_data['year_max'] == expected_met_data['year_max']
+    assert inputs_met_data['use_horizon'] == expected_met_data['use_horizon']
+    assert inputs_met_data['horizon_db'] == expected_met_data['horizon_db']
+    assert meta == META
+
+
+def test_get_pvgis_tmy_kwargs():
+    _, _, inputs, _ = get_pvgis_tmy(45, 8, usehorizon=False)
+    assert inputs['meteo_data']['use_horizon'] is False
+    data, _, _, _ = get_pvgis_tmy(
+        45, 8, userhorizon=[0, 10, 20, 30, 40, 15, 25, 5])
+    assert np.allclose(
+        data['G(h)'], USERHORIZON['G(h)'].values, equal_nan=True)
+    assert np.allclose(
+        data['Gb(n)'], USERHORIZON['Gb(n)'].values, equal_nan=True)
+    assert np.allclose(
+        data['Gd(h)'], USERHORIZON['Gd(h)'].values, equal_nan=True)
+    _, _, inputs, _ = get_pvgis_tmy(45, 8, startyear=2005)
+    assert inputs['meteo_data']['year_min'] == 2005
+    _, _, inputs, _ = get_pvgis_tmy(45, 8, endyear=2016)
+    assert inputs['meteo_data']['year_max'] == 2016
+
+
+def test_get_pvgis_tmy_basic():
+    data, _, _, _ = get_pvgis_tmy(45, 8, outputformat='basic')
+    # check each column of output separately
+    for outvar in META['outputs']['tmy_hourly']['variables'].keys():
+        assert np.allclose(data[outvar], EXPECTED[outvar], equal_nan=True)
+
+
+def test_get_pvgis_tmy_csv():
+    data, months_selected, inputs, meta = get_pvgis_tmy(
+        45, 8, outputformat='csv')
+    # check each column of output separately
+    for outvar in META['outputs']['tmy_hourly']['variables'].keys():
+        assert np.allclose(data[outvar], EXPECTED[outvar], equal_nan=True)
+    assert np.allclose(
+        [_['month'] for _ in months_selected], np.arange(1, 13, 1))
+    assert np.allclose([_['year'] for _ in months_selected], MONTHS_SELECTED)
+    assert inputs['latitude'] == INPUTS['location']['latitude']
+    assert inputs['longitude'] == INPUTS['location']['longitude']
+    assert inputs['elevation'] == INPUTS['location']['elevation']
+    for meta_value in meta:
+        if not meta_value:
+            continue
+        if meta_value == 'PVGIS (c) European Communities, 2001-2020':
+            continue
+        assert meta_value in META_VALUES
+
+
+def test_get_pvgis_tmy_epw():
+    data, _, _, meta = get_pvgis_tmy(
+        45, 8, outputformat='epw')
+    assert np.allclose(data.ghi, EXPECTED['G(h)'], equal_nan=True)
+    assert np.allclose(data.dni, EXPECTED['Gb(n)'], equal_nan=True)
+    assert np.allclose(data.dhi, EXPECTED['Gd(h)'], equal_nan=True)
+    assert np.allclose(data.temp_air, EXPECTED['T2m'], equal_nan=True)
+    assert meta == EPW_META
+
+
+def test_get_pvgis_tmy_error():
+    err_msg = 'outputformat: Incorrect value.'
+    with pytest.raises(requests.HTTPError, match=err_msg):
+        get_pvgis_tmy(45, 8, outputformat='bad')
+    with pytest.raises(requests.HTTPError, match='404 Client Error'):
+        get_pvgis_tmy(45, 8, url='https://re.jrc.ec.europa.eu/')
+
+
+META = {
```

Comment:
Took a minute to realize this was defined down here.  Not sure I like it better than a 50 line dictionary definition at the bottom of the file, but could save it as a json file and read it in at the top with the other test files.